airtable.go

```diff
@@ -92,27 +96,66 @@ func (t *GenericTable) List(opts Options) ([]Record, error) {
 	// set headsers
 	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", t.account.ApiKey))
 
+	return req, nil
+}
+
+func (t *GenericTable) doListRequest(req *http.Request) (records, error) {
 	// make request
 	client := &http.Client{}
 	resp, err := client.Do(req)
 	if err != nil {
-		return []Record{}, fmt.Errorf("Error occured: %v", err)
+		return records{}, fmt.Errorf("Error occured: %v", err)
 	}
 	defer resp.Body.Close()
 
 	// read response
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return []Record{}, fmt.Errorf("Failed to read body: %v", err)
+		return records{}, fmt.Errorf("Failed to read body: %v", err)
 	}
 
 	// unmarshal
 	ret := records{}
 	if err = json.Unmarshal(body, &ret); err != nil {
-		return []Record{}, fmt.Errorf("Failed to unmarshal response: %v", err)
+		return records{}, fmt.Errorf("Failed to unmarshal response: %v", err)
+	}
+
+	return ret, err
+}
+
+// List returns a list of records from the Airtable.
+func (t *GenericTable) List(opts Options) ([]Record, error) {
+	req, err := t.generateListRequest(opts)
+	if err != nil {
+		return []Record{}, err
```

Comment:
Oh and btw, based on the coverage report, this err check is not adequately unit tested (ie: no coverage for this line).

If you're willing, would be great if you added a test for this.